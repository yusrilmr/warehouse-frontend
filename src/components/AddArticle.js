import React, { useState } from 'react';

import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import TextField from '@material-ui/core/TextField';

const AddArticle = (props) => {
    const [open, setOpen] = useState(false);
    const [article, setArticle] = useState({ identification: '', name: '', stock: 0 });

    // Open the modal form
    const handleClickOpen = () => {
        setArticle({ identification: '', name: '', stock: 0 })
        setOpen(true);
    };

    // Close the modal form
    const handleClose = () => {
        setOpen(false);
    };

    const handleChange = (event) => {
        setArticle({...article, [event.target.name]: event.target.value});
    }

    // Save article and close modal form
    const handleSave = () => {
        props.addArticle(article);
        handleClose();
    }

    return (
        <div>
            <Button variant="outlined" color="primary" style={{margin: 10}} onClick={handleClickOpen}>
                New Article
            </Button>
            <Dialog open={open} onClose={handleClose}>
                <DialogTitle>New Article</DialogTitle>
                <DialogContent>
                    <TextField autoFocus fullWidth label="Identification" name="identification"
                               value={article.identification} onChange={handleChange}/>
                    <TextField fullWidth label="Name" name="name"
                               value={article.name} onChange={handleChange}/>
                    <TextField fullWidth label="Stock" name="stock"
                               value={article.stock} onChange={handleChange}/>
                </DialogContent>
                <DialogActions>
                    <Button color="secondary" onClick={handleClose}>Cancel</Button>
                    <Button color="primary" onClick={handleSave}>Save</Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};
export default AddArticle;
AddArticle.propTypes = {
    addArticle: PropTypes.func
};