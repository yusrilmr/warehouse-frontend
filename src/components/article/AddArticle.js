import React, { useState } from 'react';

import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import TextField from '@material-ui/core/TextField';

const AddArticle = (props) => {
    const [open, setOpen] = useState(false);
    // const [errorText, setErrorText] = useState({ identification: '', name: '', stock: '' });
    const [errorIdentificationText, setErrorIdentificationText] = useState('');
    const [errorNameText, setErrorNameText] = useState('');
    const [errorStockText, setErrorStockText] = useState('');

    const [article, setArticle] = useState({ identification: '', name: '', stock: 0 });

    const emptyErrorText = () => {
        // setErrorText({ identification: '', name: '', stock: '' })
        setErrorIdentificationText('');
        setErrorNameText('');
        setErrorStockText('');
    };

    // Open the modal form
    const handleClickOpen = () => {
        emptyErrorText();
        setArticle({ identification: '', name: '', stock: 0 })
        setOpen(true);
    };

    // Close the modal form
    const handleClose = () => {
        setOpen(false);
    };

    const handleChange = (event) => {
        setArticle({...article, [event.target.name]: event.target.value});
    };

    // Save article and close modal form
    const handleSave = () => {
        if (isValid()) {
            props.addArticle(article);
            handleClose();
        }
    };

    const isValid = () => {
        emptyErrorText();
        let valid = true;
        if (article.identification === '') {
            // setErrorText({...errorText, identification: "This field is mandatory"});
            setErrorIdentificationText("This field is mandatory");
            valid = false;
        }
        if (article.name === '') {
            // setErrorText({...errorText, name: "This field is mandatory"});
            setErrorNameText("This field is mandatory")
            valid = false;
        }
        if (article.stock < 0) {
            // setErrorText({...errorText, stock: "This field cannot be less than 0"});
            setErrorStockText("This field cannot be less than 0")
            valid = false;
        }
        return valid;
    };

    return (
        <div>
            <Button variant="outlined" color="primary" style={{margin: 10}} onClick={handleClickOpen}>
                New Article
            </Button>
            <Dialog open={open} onClose={handleClose}>
                <DialogTitle>New Article</DialogTitle>
                <DialogContent>
                    <TextField autoFocus fullWidth label="Identification" name="identification"
                               value={article.identification} onChange={handleChange}
                               error ={!!errorIdentificationText.length} helperText={errorIdentificationText}/>
                    <TextField fullWidth label="Name" name="name"
                               value={article.name} onChange={handleChange}
                               error ={!!errorNameText.length} helperText={errorNameText}/>
                    <TextField fullWidth label="Stock" name="stock" type="number"
                               value={article.stock} onChange={handleChange}
                               error ={!!errorStockText.length} helperText={errorStockText}/>
                </DialogContent>
                <DialogActions>
                    <Button color="secondary" onClick={handleClose}>Cancel</Button>
                    <Button color="primary" onClick={handleSave}>Save</Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};
export default AddArticle;
AddArticle.propTypes = {
    addArticle: PropTypes.func
};