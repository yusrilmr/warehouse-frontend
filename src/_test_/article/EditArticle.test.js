import React from 'react';
import AddArticle from '../../components/article/AddArticle';
import TextField from '@material-ui/core/TextField';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<AddArticle />', () => {
    it('renders three <TextInput /> components', () => {
        const wrapper = shallow(<AddArticle />);
        expect(wrapper.find(TextField)).toHaveLength(3);
    });
});