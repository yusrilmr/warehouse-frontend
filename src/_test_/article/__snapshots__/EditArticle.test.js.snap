// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      New Article
    </span>
  </button>
</div>
`;
